### ISSUE TITLE
Feature: Multivariate Normal with sparse precision matrix

### ISSUE THREAD

**Issue by Garoe** [tensorflow/probability#895]
I'm the author of https://github.com/Garoe/tf_mvg/ and I've been considering writing a pull request for the Multivariate Norma distribution with sparse precision matrix described there. However, before starting any work I wanted to ensure that this is something that is likely to be merged. The main issue is that for any interesting image size any operation that requires the covariance matrix (most notably, sampling from the distribution) is not computationally tractable in tensorflow as sparse solvers are not supported yet. Hence, that was implemented using scipy sparse solvers on CPU without any gradient information. Even that is not enough for larger image sizes, so a modified sparse solver that trades memory for speed is also provided.

So, I see how that is quite problematic as it is a breach of the first point in the Distribution class contract
https://github.com/tensorflow/probability/blob/master/discussion/tfp_distributions_contract.md

**Comment by langmore** [comment]
You may want to try [`MultivariateNormalPrecisionFactorLinearOperator`](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/experimental/distributions/mvn_precision_factor_linop.py)  You can optionally initialize this MVN with a `LinearOperator` that defines the precision. This will allow effiicent log prob computations.  Note that there is no current sparse `LinearOperator`, so this only helps if the sparse structure you need is found in one of [the existing operators](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/ops/linalg).

**Comment by Garoe** [comment]
Thanks for the reply. The code I wrote in that repository already includes a very efficient log prob computation by using the convolution operator, as that is the sparsity structure that was used for the Cholesky matrix. However, as I wrote on my first message, without an sparse solver, which is still the case for tensorflow, sampling from the distribution cannot be done efficiently. Anyhow, I'm no longer available for implementing this and I don't see much interest either, so I'm closing this issue.
